tests/postgres_tests/test_array.py

```diff
@@ -363,6 +365,25 @@ class Meta:
         self.assertEqual(len(errors), 1)
         self.assertEqual(errors[0].id, 'postgres.E002')
 
+    def test_nested_field_checks(self):
+        """
+        #25867 - Nested ArrayFields are permitted.
+        """
+        test_apps = Apps(['postgres_tests'])
+
+        class MyModel(PostgreSQLModel):
+            field = ArrayField(ArrayField(models.CharField()))
+
+            class Meta:
+                apps = test_apps
+
+        model = MyModel()
+        errors = model.check()
+        self.assertEqual(len(errors), 1)
+        # Base field error messages are wrapped in 'postgres.E001'
```

Comment:
I guess I would say something like "The inner CharField is missing a max_length."